pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -226,6 +226,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.gcResourceTrackers(logCtx, handler, common.ApplicationRendering, false)
 	case common.WorkflowStateSuspended:
 		logCtx.Info("Workflow return state=Suspend")
+		if !workflow.IsFailedAfterRetry(app) {
+			r.stateKeep(logCtx, handler, app)
```

Comment:
does the stateKeep executed once or periodically?

Reply:
Periodically. The gcResourceTracker return will use ApplicationReSyncPeriod here, which means in suspending status, application will requeue after ApplicationReSyncPeriod.